src/Tools/PrepareTests/TestDiscovery.cs

```diff
@@ -71,73 +72,35 @@ public static bool RunDiscovery(string repoRootDirectory, string dotnetPath, boo
+    static (bool Succeeded, string Output) RunWorker(string dotnetPath, string pathToWorker, string pathToAssembly)
     {
-        var success = true;
-        var pipeClient = new Process();
-        var arguments = new List<string>();
+        var worker = new Process();
+        var arguments = new StringBuilder();
         if (pathToWorker.EndsWith("dll"))
         {
-            arguments.Add(pathToWorker);
-            pipeClient.StartInfo.FileName = dotnetPath;
+            arguments.Append($"exec {pathToWorker}");
+            worker.StartInfo.FileName = dotnetPath;
         }
         else
         {
-            pipeClient.StartInfo.FileName = pathToWorker;
+            worker.StartInfo.FileName = pathToWorker;
         }
 
-        var errorOutput = new StringBuilder();
-
-        using (var pipeServer = new AnonymousPipeServerStream(PipeDirection.Out, HandleInheritability.Inheritable))
-        {
-            // Pass the client process a handle to the server.
-            arguments.Add(pipeServer.GetClientHandleAsString());
-            pipeClient.StartInfo.Arguments = string.Join(" ", arguments);
-            pipeClient.StartInfo.UseShellExecute = false;
-
-            // Errors will be logged to stderr, redirect to us so we can capture it.
-            pipeClient.StartInfo.RedirectStandardError = true;
-            pipeClient.ErrorDataReceived += PipeClient_ErrorDataReceived;
-            pipeClient.Start();
-
-            pipeClient.BeginErrorReadLine();
-
-            pipeServer.DisposeLocalCopyOfClientHandle();
-
-            try
-            {
-                // Read user input and send that to the client process.
-                using var sw = new StreamWriter(pipeServer);
-                sw.AutoFlush = true;
-                // Send a 'sync message' and wait for client to receive it.
-                sw.WriteLine("ASSEMBLY");
-                // Send the console input to the client process.
-                sw.WriteLine(pathToAssembly);
-            }
-            // Catch the IOException that is raised if the pipe is broken
-            // or disconnected.
-            catch (Exception e)
-            {
-                Console.Error.WriteLine($"Error: {e.Message}");
-                success = false;
-            }
-        }
-
-        pipeClient.WaitForExit();
-        success &= pipeClient.ExitCode == 0;
-        pipeClient.Close();
-
-        if (!success)
-        {
-            Console.WriteLine($"Failed to discover tests in {pathToAssembly}:{Environment.NewLine}{errorOutput}");
-        }
-
-        return success;
-
-        void PipeClient_ErrorDataReceived(object sender, DataReceivedEventArgs e)
-        {
-            errorOutput.AppendLine(e.Data);
-        }
+        var pathToOutput = Path.Combine(Path.GetDirectoryName(pathToAssembly)!, "testlist.json");
+        arguments.Append($" --assembly {pathToAssembly} --out {pathToOutput}");
+
+        string output = string.Empty;
```

Comment:
nit: feels slightly strange that we go to the trouble of a StringBuilder for the arguments but not for this.

Reply:
Got a bit lazy cause I knew the output was a single line. will use a builder.